src/drafty.js

```diff
@@ -2236,11 +2272,13 @@ function addNode(parent, n) {
 }
 
 // Returns a tree of nodes.
-function spansToTree(parent, text, start, end, spans) {
+function spansToTree(parent, graphemes, start, end, spans) {
   if (!spans || spans.length == 0) {
     if (start < end) {
       addNode(parent, {
-        text: text.substring(start, end)
+        text: graphemes.slice(start, end)
+          .map(segment => segment.segment)
+          .join("")
```

Comment:
Code style: please use ' (single quotes) for strings which require no localization.